release 23.4 notes (#381)
stackabletech/documentation: modules/ROOT/pages/release_notes.adoc

The Stackable platform consists of multiple operators that work together.
Periodically a platform release is made, including all components of the platform at a specific version.

== Release 23.4

The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been adapted and tested for compatibility with Openshift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against Openshift: over the coming weeks this certification process will now be extended across all operators and will be specific to each platform-release.

=== New/extended platform features
The following new major platform features were added:

Status Field::

> Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use the several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.

Author: That should work when the docs are taken live.

Log Aggregation::

The logging framework (added to the platform in Release 23.1) that offers a consistent custom resource configuration and a separate, persisted sink, has been rolled out across all operators. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR023-product-image-selection.html[ADR] and this https://docs.stackable.tech/home/stable/concepts/product_image_selection.html[concepts page] for more information.

Service Type::

> The Service type can now be specified: this enables a later switch to using ListenerClasses for the exposure of Services. This change is breaking: for security reasons the default is set to the cluster-internal ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable or external-stable.

Reviewer: ```suggestion
The Service type can now be specified in all products. This currently differences between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change is breaking. For security reasons the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).
```

* https://github.com/stackabletech/secret-operator/issues/4[Secret operator: Kerberos keytab provisioning]

=== stackablectl

The following have been added to `stackablectl`:

==== Trino-iceberg demo

This is a condensed form of the https://docs.stackable.tech/stackablectl/stable/demos/data-lakehouse-iceberg-trino-spark.html[data-lakehouse-iceberg-trino-spark] demo focusing on using the lakehouse to store and modify data. It demonstrates how to integrate Trino and Iceberg and should run on a local workstation.

==== Jupyterhub/Spark demo

This demo showcases the integration between Jupyter and Apache Hadoop deployed on the Stackable Data Platform (SDP) Kubernetes cluster. This demo can be installed on most cloud managed Kubernetes clusters as well as on premise or on a reasonably provisioned laptop.

The xref:stackablectl::quickstart.adoc[quickstart guide] shows how to get started with `stackablectl`. This link lists the xref:stackablectl::demos/index.adoc[available demos].

=== Supported Kubernetes versions
This release supports the following Kubernetes versions:

* `1.26`
* `1.25`
* `1.24`
* `1.23`

==== Breaking changes

> You will need to adapt your existing CRDs due to the following breaking changes detailed below.

Author: Yes, good point. I'll leave 1.23 in but qualify its support.

The Stackable platform consists of multiple operators that work together.
Periodically a platform release is made, including all components of the platform at a specific version.

== Release 23.4

> The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been adapted and tested for compatibility with Openshift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against Openshift. Over the coming weeks this certification process will now be extended across all operators and will be specific to each platform-release.

Reviewer 2: I removed the "what we will do in the future" stuff from here.

```suggestion
The focus in this platform release is on the support of default/custom affinities and the status field, as well as the rollout of log aggregation across the remaining operators. Additionally, all operators have been updated and tested for compatibility with OpenShift clusters (versions 4.10 and 4.11). Several operators from the 23.1 platform release were already certified against OpenShift.
```

=== New / extended platform features
The following new major platform features were added:

Cluster Operation::

The first part of https://docs.stackable.tech/home/stable/concepts/cluster_operations.html[ClusterOperations] was rolled out in every applicable Stackable Operator. This supports pausing the cluster reconciliation and stopping the cluster completely. Pausing reconciliation will not apply any changes to the Kubernetes resources (e.g. when changing the custom resource). Stopping the cluster will set all replicas of StatefulSets, Deployments or DaemonSets to zero and therefore deleting all Pods belonging to that cluster (not the PVCs).

Status Field::

> Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use the several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.

Reviewer 2: ```suggestion
Operators of the Stackable Data Platform create, manage and delete Kubernetes resources: in order to easily query the health state of the products - and react accordingly - Stackable Operators use several predefined condition types to capture different aspects of a product's availability. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR027-status[ADR] for more information.
```

Default / Custom Affinities::

> In Kubernetes there are different ways to influence how Pods are assigned to Nodes. In some cases it makes sense to co-locate certain services that communicate a lot with each other, such as HBase regionservers with HDFS datanodes. In other cases it makes sense to distribute the Pods among as many Nodes as possible. There may also be additional requirements e.g. placing important services - such as HDFS namenodes - in different racks, datacenter rooms or even datacenters. This release supports default affinities that should suffice for many scenarios out-of-the box, as well as enabling the configuration of custom affinity rules at a role and/or role-group level. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR026-affinities[ADR] for more information.

Reviewer 2: ```suggestion
In Kubernetes there are different ways to influence how Pods are assigned to Nodes. In some cases it makes sense to co-locate certain services that communicate a lot with each other, such as HBase regionservers with HDFS datanodes. In other cases it makes sense to distribute the Pods among as many Nodes as possible. There may also be additional requirements e.g. placing important services - such as HDFS namenodes - in different racks, datacenter rooms or even datacenters. This release implements default affinities that should suffice for many scenarios out-of-the box, while also allowing for custom affinity rules at a role and/or role-group level. See this https://docs.stackable.tech/home/stable/contributor/adr/ADR026-affinities[ADR] for more information.
```

Log Aggregation::

> The logging framework (added to the platform in Release 23.1) offers a consistent custom resource configuration and a separate, persisted sink (the default implementation is Opensearch but is interchangeable). This has now been rolled out across all operators. See this https://docs.stackable.tech/home/stable/contributor/adr/adr025-logging_architecture[ADR] and this https://docs.stackable.tech/home/stable/concepts/logging.html[concepts page] for more information.

Reviewer 2: ```suggestion
The logging framework (added to the platform in Release 23.1) offers a consistent custom resource configuration and a separate, persisted sink (defaulting to OpenSearch). This has now been rolled out across all products. See this https://docs.stackable.tech/home/stable/contributor/adr/adr025-logging_architecture[ADR] and this https://docs.stackable.tech/home/stable/concepts/logging.html[concepts page] for more information.
```

Service Type::

> The Service type can now be specified in all products. This currently differentiates between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change is breaking. For security reasons the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be made accessible from outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).

Reviewer 2: ```suggestion
The Service type can now be specified in all products. This currently differentiates between the internal ClusterIP and the external NodePort and is forward compatible with the [ListenerClass](https://docs.stackable.tech/home/stable/listener-operator/listenerclass.html) for the automatic exposure of Services via the Listener Operator. This change not backwards compatible with older platform releases. For security reasons, the default is set to the cluster-internal (ClusterIP) ListenerClass. A cluster can be exposed outside of Kubernetes by setting clusterConfig.listenerClass to external-unstable (NodePort) or external-stable (LoadBalancer).
```

* https://github.com/stackabletech/secret-operator/issues/4[Secret operator: Kerberos keytab provisioning]

=== stackablectl

The following have been added to `stackablectl`:

==== Trino-iceberg demo

This is a condensed form of the https://docs.stackable.tech/stackablectl/stable/demos/data-lakehouse-iceberg-trino-spark.html[data-lakehouse-iceberg-trino-spark] demo focusing on using the lakehouse to store and modify data. It demonstrates how to integrate Trino and Iceberg and should run on a local workstation.

==== Jupyterhub/Spark demo

This demo showcases the integration between Jupyter and Apache Hadoop deployed on the Stackable Data Platform (SDP) Kubernetes cluster. This demo can be installed on most cloud managed Kubernetes clusters as well as on premise or on a reasonably provisioned laptop.

The xref:stackablectl::quickstart.adoc[quickstart guide] shows how to get started with `stackablectl`. This link lists the xref:stackablectl::demos/index.adoc[available demos].

=== Supported Kubernetes versions

This release supports the following Kubernetes versions:

* `1.26`
* `1.25`
* `1.24`

> * `1.23` (it is planned to discontinue support for this version in the next release)

Reviewer 2: Maybe also list OpenShift 4.10 and 4.11 now?
